endpoints/v2/manifest.py

```diff
@@ -47,129 +49,158 @@
 MANIFEST_TAGNAME_ROUTE = BASE_MANIFEST_ROUTE.format(VALID_TAG_PATTERN)
 
 
-@v2_bp.route(MANIFEST_TAGNAME_ROUTE, methods=["GET"])
-@disallow_for_account_recovery_mode
-@parse_repository_name()
-@process_registry_jwt_auth(scopes=["pull"])
-@require_repo_read
-@anon_protect
-def fetch_manifest_by_tagname(namespace_name, repo_name, manifest_ref):
-    repository_ref = registry_model.lookup_repository(namespace_name, repo_name)
-    if repository_ref is None:
-        image_pulls.labels("v2", "tag", 404).inc()
-        raise NameUnknown()
+class FetchManifest(MethodView):
```

Comment:
Essentially the same comment as before on the blob class endpoints. I'd prefer to keep it consistent across the registry api, unless there's a reason this should be wrapped in a class.

Reply:
This is not the same as for the blob class endpoints. Inheritance is in use here to share proxy code between the two manifest GET endpoints. I didn't know we had an unspoken agreement towards prefering functional programming over object oriented - if we do I can happilly adhere.